### PR TITLE
Feature/ds attack app

### PR DESCRIPTION
**Pull request by this-kramer**
## Reason for this PR:

- We can now perform a double-spending attack in the app
- The following UI is hidden in the context menu that allows discarding updated tokens and hence reusing the olds ones

ds

### CODE REVIEW

File: android/app/src/main/java/org/cryptimeleon/incentive/app/NavGraph.kt [hunk 1]
```diff
@@ -15,6 +15,7 @@ import androidx.navigation.NavType
 import androidx.navigation.compose.NavHost
 import androidx.navigation.compose.composable
 import androidx.navigation.navArgument
+import org.cryptimeleon.incentive.app.ui.attacker.AttackerUi
```
Comment (pschuermann97) [inline android/app/src/main/java/org/cryptimeleon/incentive/app/NavGraph.kt:18]: AttackerUI is the UI you show on the PR page, I assume

File: android/app/src/main/java/org/cryptimeleon/incentive/app/ui/basket/BasketViewModel.kt [hunk 2]
```diff
@@ -73,12 +46,4 @@ class BasketViewModel @Inject constructor(
...
         }
     }
-
-    fun onDiscardClicked() {
-        viewModelScope.launch {
-            withContext(Dispatchers.IO) {
-                basketRepository.discardCurrentBasket(true)
-            }
-        }
-    }
```
Comment (pschuermann97) [inline android/app/src/main/java/org/cryptimeleon/incentive/app/ui/basket/BasketViewModel.kt:83]: why so many deleted lines in the basket view model?

**Review by pschuermann97 — APPROVED** [review]

File: android/app/src/main/java/org/cryptimeleon/incentive/app/ui/basket/BasketViewModel.kt (hunk 2 above)
Comment (this-kramer) [reply android/app/src/main/java/org/cryptimeleon/incentive/app/ui/basket/BasketViewModel.kt:83]: Forgot to delete it earlier

File: android/app/src/main/java/org/cryptimeleon/incentive/app/NavGraph.kt (hunk 1 above)
Comment (this-kramer) [reply android/app/src/main/java/org/cryptimeleon/incentive/app/NavGraph.kt:18]: Indeed it is

**Pull request merged**